clang/test/SemaSYCL/sycl-device-intel-fpga-local.cpp

```diff
@@ -0,0 +1,24 @@
+// RUN: %clang_cc1 -fsycl -fsycl-is-device -fsyntax-only -ast-dump -verify -pedantic %s | FileCheck %s
```

Comment:
Sorry I just noticed the name of this test. Is there a reason you didn't add it to intel-fpga-local.cpp? I think you can add these tests there. You can AST check there for attribute on global constant variable (if a test does not already exist), or rename this file to intel-max-replicates-global-const.cpp or something. I prefer the former since it looks like all FPGA attributes are being tested there (including other max_replicates tests).

Reply:
>>Is there a reason you didn't add it to intel-fpga-local.cpp? 

Inside intel-fpga-local.cpp file, all FPGA attributes apply on local variables. The name of the test represents this.

>>You can AST check there for attribute on global constant variable (if a test does not already exist), 

We need to rename this test then since there is no tests for global constant variables. I will prefer not to rename the test and add there since PrivateCopies attribute does not support global constant variables. Many be we have other Decl attributes that do not support global constant variables (i have not checked this ).

 >>clang/test/SemaSYCL/sycl-device-intel-fpga-local.cpp

I will prefer to rename this test to intel-fpga-global.cpp or something unique if that sounds good to @elizabethandrews  instead of intel-max-replicates-global-const.cpp because i plan to refactor another Decl attributes that support global constant variables. So we can use this test for other attributes in feature.

Reply:
@elizabethandrews, i have renamed this test to intel-fpga-global-const.cpp so that we can use the same test for other attributes on global constant variables. . 

we will refactor this generic function: AddOneConstantPowerTwoValueAttr()  and AddOneConstantValueAttr() in future.
The attributes below all support this, we can use the same test.
IntelFPGANumBanks
IntelFPGABankWidth
IntelFPGABankBits
IntelFPGAForcePow2Depth

Reply:
Ok Thanks for clarification!